manifests/init.pp

```diff
@@ -126,6 +126,7 @@
   $wait_for_pxp_agent_exit = undef,
   $wait_for_puppet_run     = undef,
   Array[Puppet_agent::Config] $config = [],
+  $version_file_path       = $facts['os']['family'] ? { 'windows' => 'C:\Program Files\Puppet Labs\Puppet\VERSION', default => '/opt/puppetlabs/puppet/VERSION' }
```

Comment:
On Windows the install dir is user-configurable/relocatable, but there's a core fact you can use https://www.puppet.com/docs/puppet/7/core_facts.html#env-windows-installdir

```suggestion
  $version_file_path       = $facts['os']['family'] ? { 'windows' => "${facts['env_windows_installdir']}\\VERSION", default => '/opt/puppetlabs/puppet/VERSION' }
```
One tricky thing, you'll need to double escape the backslash in the double quoted string (since we're interpolating). I don't know if that syntax works exactly, but something like it should.

Reply:
You can also use forward slashes on Windows.